File: app/src/main/java/com/abishov/hexocat/home/repository/RepositoryViewModel.java
package com.abishov.hexocat.home.repository;

import android.os.Parcelable;
import com.google.auto.value.AutoValue;

@AutoValue
public abstract class RepositoryViewModel implements Parcelable {

  public static RepositoryViewModel create(String name, String description,
      String forks, String stars, String ownerAvatar, String login, String url) {
    return new AutoValue_RepositoryViewModel(name, description, forks,
        stars, ownerAvatar, login, url);
  }

  public abstract String name();

  public abstract String description();

  public abstract String forks();

  public abstract String stars();

  public abstract String avatarUrl();

  public abstract String login();

  public abstract String url();
}

File: hexocat_android/app/src/main/java/com/abishov/hexocat/composables/RepositoryViewModel.kt
package com.abishov.hexocat.composables

import com.abishov.hexocat.composables.LanguageViewModel
import com.abishov.hexocat.composables.TopicViewModel

data class RepositoryViewModel(
  val name: String,
  val description: String,
  val forks: String,
  val stars: String,
  val avatarUrl: String,
  val bannerUrl: String,
  val usesBannerUrl: Boolean,
  val login: String,
  val url: String,
  val languages: List<LanguageViewModel>,
  val topics: List<TopicViewModel>
)
